src/java.desktop/share/classes/java/awt/SplashScreen.java

```diff
@@ -208,33 +210,28 @@ private void checkVisible() {
             throw new IllegalStateException("no splash screen available");
         }
     }
+
     /**
      * Returns the current splash screen image.
      *
      * @return URL for the current splash screen image file
      * @throws IllegalStateException if the splash screen has already been closed
      */
-    @SuppressWarnings("deprecation")
     public URL getImageURL() throws IllegalStateException {
         synchronized (SplashScreen.class) {
             checkVisible();
             if (imageURL == null) {
                 try {
                     String fileName = _getImageFileName(splashPtr);
-                    String jarName = _getImageJarName(splashPtr);
                     if (fileName != null) {
-                        if (jarName != null) {
-                            @SuppressWarnings("deprecation")
-                            var _unused = imageURL =
-                                    new URL("jar:"+(new File(jarName).toURL().toString())+"!/"+fileName);
-                        } else {
-                            imageURL = new File(fileName).toURL();
-                        }
+                        String jarName = _getImageJarName(splashPtr);
+                        // When calling java -jar, point to the file within the jar
+                        String schema = jarName != null ? "jar:" + jarName + "!/" : "file:";
+                        imageURL = new URI(schema + fileName).toURL();
```

Comment:
This looks broken, I would expect to see File(fileName).toURI() or Path.of(fileName).toUri() to get the URI.

Reply:
Tried the suggestions, and they are both equivalent (ie `.equals`) to the one in the pull. Would there be something to be concerned about doing it this way?

Reply:
The input for the URI constructor is a URI string, not a file path. Converting from a File (or Path) to a URI requires deal with escaping issues, replacing slashes, etc. You may have to expand your testing to see the issues.